src/game.cpp

```diff
@@ -5742,6 +5758,13 @@ int game::assign_faction_id()
     return ret;
 }
 
+long game::assign_creature_id()
+{
+    long ret = next_creature_id;
```

Comment:
Alternately you can take the nuclear option: https://github.com/graeme-hill/crossguid.
Given the fact that we're potentially generating really huge numbers of entities AND dealing with concurrency, it's not actually THAT overkill.